src/filesystem/file_content.rs

```diff
@@ -150,10 +160,20 @@ impl FileContent {
         matches!(self, Self::Stub { .. })
     }
 
+    pub fn is_empty(&self) -> bool {
+        match self {
+            Self::EmptyFile => true,
+            Self::Encrypted { data_size, .. } => *data_size < 1u64,
+            Self::Public { data_size, .. } => *data_size < 1u64,
+            Self::Stub { .. } => true,
```

Comment:
Stub should probably also be based on its internal data_size (its not exactly empty, its just not yet available) but I think this is a semantic argument that doesn't matter (you don't have to change anything hahah).

Reply:
Liberty had the same dialogue in my mind. I'm trying to convey the idea that the files not readable yet ( read() being the main consumer of that method)